### PR TITLE
Email validator in flutter

### LINKED ISSUE #3932: How To Validate Emails in Flutter
We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner. Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles to go over.

**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**NOTE: We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

*We're looking for the first 2-3 paragraphs of the article that appropriately summarize what your article will be about.*

### How To Validate Emails in Flutter
Your title should be descriptive of the article/tutorial. Be Specific. *Use keyword research to gain your article higher ranking.*
**Use the following format as the title of the issue: [Category of Topic] Title for the article.**

For example, if the article is based on machine learning, use the following: [Machine learning] Introduction to Machine Learning.
If the article is based on developing an Android application, use the following: [Android] Developing Apps using Android.

ONLY viable topics to pick from:
- API
- Artificial Intelligence
- Blockchain
- Containers
- Edge Computing
- Languages
- Machine Learning
- Networking
- Node.js
- Security

### Introduction paragraph (2-3 paragraphs):
Email validator is a Dart class for validating email addresses without using Reg Ex. It Can also be used to validate emails within Flutter apps. It provides you with one easy-to-use function to validate an email input. The author mentions that this is a simple (but “correct”) Dart class for email validation without using regular expressions. We’ll come back later to why he calls it correct.

I think the purpose of this package is very clear. For example, if you want to register a user with an email address and a password, you have to verify that the email address is valid. If it’s not, you can give your user a hint that they have to check their email address again for any spelling mistakes. Not all backend systems support email validation while creating a new user. According to your app’s user experience, it’s also much better to help your user when they’ve finished the email address input and want to start filling in a password.

### Key takeaways: 
- learn how to validate emails in flutter.
- coding the email validation  in flutter.
- understand the implementation of email validator.

### References: 
- https://pub.dev/documentation/email_validator/latest/
- https://betterprogramming.pub/how-to-validate-emails-in-flutter-957ae75926c9

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by joni-kano**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #3932

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by joni-kano** [comment]
@jerimkaura @Kayere I have polished the article

**Comment by WanjaMIKE** [comment]
@jerimkaura could you take  up this PR. It appears you were reviewing it before. Thanks

**Comment by LinusMuema** [comment]
plagiarism check done :heavy_check_mark:

**Comment by hectorkambow** [comment]
@jerimkaura @Kayere  

Can we get a status on this PR. 

If we need to get this re-assigned we can do so 👍

**Comment by hectorkambow** [comment]
@joni-kano 
Good afternoon and thank you for submitting your article for publishing. With the recent increase in article volume out team has been working diligently to review every article. 

Your article has been entered into our queue and should be reviewed (and published pending any changes) as soon as a peer reviewer is finished with the ones in the queue before it. 🚀
Thank you.

**Comment by hectorkambow** [comment]
Peer reviewer pinged via Slack as well :thumb

File: content/articles/email-validator-in-flutter/index.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+- Understand the implementation of email validator.
+
+### Working mechanism of email validator
+Email validator runs a check on the emails through a process called `swift process`. This process detects typographical errors or deliberate mistakes. Furthermore, it helps to organize and clean all email addresses in a flutter application.
+
+For this reason, it protects the email sender score while maximizing the efficiency of the flutter application. The key concepts from the use of this package are validity, risk, and the invalidity of any email address.
+
+This package consists of a class and a function. 
+
+The constructor for this class is pretty simple: Create a new instance of this class and set the validator type to either 1 or 2 (defaults to 1). The constructor should pass a string and an array of strings for validation.The arguments are also key factors.
```
Comment (jerimkaura) [inline content/articles/email-validator-in-flutter/index.md:32]: Explain type 1 and 2 and what each type does.

File: content/articles/email-validator-in-flutter/index.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+### Key takeaways
+- What is an email validator?
+- Explaining how an email validator works.
+- Create a simple app to show email validator.
+- Understand the implementation of email validator.
+
+### Working mechanism of email validator
+Email validator runs a check on the emails through a process called `swift process`. This process detects typographical errors or deliberate mistakes. Furthermore, it helps to organize and clean all email addresses in a flutter application.
+
+For this reason, it protects the email sender score while maximizing the efficiency of the flutter application. The key concepts from the use of this package are validity, risk, and the invalidity of any email address.
```
Comment (jerimkaura) [inline content/articles/email-validator-in-flutter/index.md:28]: Explain the key concepts

File: content/articles/email-validator-in-flutter/index.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+This package consists of a class and a function. 
+
+The constructor for this class is pretty simple: Create a new instance of this class and set the validator type to either 1 or 2 (defaults to 1). The constructor should pass a string and an array of strings for validation.The arguments are also key factors.
+
+The first argument is self-explanatory. It is just a boolean that specifies whether we are using the `local-part@domain.com` format or `domain.com` as our string to validate.
+
+The second argument allows you to pass an array of strings that are valid domains for the email. This argument is only necessary if the string passed in as our first argument is in `domain.com` format; otherwise, it can be omitted.
+
+If either of the first two arguments contains an error, then false should be returned. Otherwise, true should be returned, and our string should be validated.
```
Comment (jerimkaura) [inline content/articles/email-validator-in-flutter/index.md:38]: Where are the arguments? Explain them in detail with a proper example to enable understanding of the concept.

File: content/articles/email-validator-in-flutter/index.md [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+
+### Benefits of email validator
+1. Makes sure the user enters a correctly formatted email address while signing up.
+2. Ensures emails are not on the blocklist and are not sent to blacklisted email addresses or domains.
+3. Alerts if the username is available on the website during the registration process.
+4. You can add it as a pre-requisite while submitting the app to the app store and google play store for review purposes.
```
Comment (jerimkaura) [inline content/articles/email-validator-in-flutter/index.md:44]: Explain how each benefit is acheived.

File: content/articles/email-validator-in-flutter/index.md [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+
+### Areas of application
+- Address CaptureSingle-line Address Verification Software.
+- Address VerifyAddress Enhancement.
+- Data CleanseBatch Data Cleanse & Maintenance.
+- Bank VerificationBank Account, Sort Code & IBAN Validation.
+- GeocodeLongitude & Latitude Data.
+- Email ValidationCheck and Verify Email Addresses.
+- This example brings us to the understanding usage of this API for serverless flutter applications.
```
Comment (jerimkaura) [inline content/articles/email-validator-in-flutter/index.md:53]: Provide a detailed explanation of how email validation works in each of the specified areas. What module of the system implemented it. At what level does it work. For instance, answer the question How is `GeocodeLongitude & Latitude Data` applying this concept?

File: content/articles/email-validator-in-flutter/index.md [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+
+### Benefits of using Email Validator.
+Email validator works better in that it is a more accurate validation tool. In addition to validating email addresses, it also checks that domains are not likely to exist soon.
```
Comment (jerimkaura) [inline content/articles/email-validator-in-flutter/index.md:56]: More accurate than........? And will it determine that a domain will not exist soon?

File: content/articles/email-validator-in-flutter/index.md [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+
+Domain Validation Tool is a free service that allows you to search up to 500 domains from your list for availabilitydor and immediate results. It's the most potent domain availability checker available on the internet today. You can quickly check up to 500 domains from your list for availability and receive immediate results.
```
Comment (jerimkaura) [inline content/articles/email-validator-in-flutter/index.md:58]: This is the first time you mention this in the article and it looks new. How will a reader connect it to the parent concept being discussed?

File: content/articles/email-validator-in-flutter/index.md [hunk 8]
```diff
@@ -0,0 +1,238 @@
...
+
+|                                       Valid email input                                       |                                         After running and valid email                                         |
+| :-------------------------------------------------------------------------------------------: | :-----------------------------------------------------------------------------------------------------------: |
+| ![Valid Email Input](/engineering-education/flutter-email-validator/validemail.jpg) | ![After running and valid email](/engineering-education/flutter-email-validator/validatedemail.jpg) |
+
+### Conclusion
+Email validator is one of the best ways to confirm emails in Flutter. It helps to reduce the invalid emails that users may use to sign up and help one easily maintain the available emails.
+
+I recommend any programmer to use any flutter application due to its' compatibility with all levels of flutter application, be it flutter web application, flutter mobile application, or any other application.
```
Comment (jerimkaura) [inline content/articles/email-validator-in-flutter/index.md:232]: A conclusion should summarize your article as well. What can be learned from your article is the conclusion here.

**Review by jerimkaura — CHANGES_REQUESTED** [review]
Hello @joni-kano 

A quick one. Take the corrections I have suggested in the comments and address them in detail. Remember it is just about a working code, we need to have a great flow and easy-to-read content as well.

Your article seems to have a lot of readability issues. Furnish it then ping me here or on slack @jerimkaura

**Comment by joni-kano** [comment]
Noted with thanks.i am working on it

**Comment by joni-kano** [comment]
@jerimkaura I have revised the article as you would want

**Comment by hectorkambow** [comment]
> @jerimkaura I have revised the article as you would want

@joni-kano 

Please be sure you are going over the entire article as our peer reviewers are not responsible to point out every mistake - that is the authors responsibility. 

If any more major revisions are needed - we will be closing this PR asap. 👀 

cc @jerimkaura

**Comment by joni-kano** [comment]
Acknowledged.

**Comment by hectorkambow** [comment]
reviewer pinged via slack

**Comment by jerimkaura** [comment]
Hello @joni-kano 

Seem like your article still has a lot of readability issues. I will be closing this PR. Feel free to write on another topic or polish this until it's ready for publishing.

**Comment by joni-kano** [comment]
I will polish this until it's ready

**Pull request closed without merging**